.github/workflows/nightly.yml

```diff
@@ -0,0 +1,125 @@
+name: Nightly
+
+on:
+  schedule:
+    - cron: 0 0 * * *
+
+
+jobs:
+  generate-proving-key:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: dev
+
+      - name: Use Node.js 16
+        uses: actions/setup-node@v4
+        with:
+          node-version: 16
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install --yes \
+            build-essential \
+            libgmp-dev \
+            libsodium-dev \
+            nasm \
+            nlohmann-json3-dev
+
+      # To prevent `npm install` failure of circuit package, 
+      # it has to checkout manually because version of `circomlib` has pinned to a specific commit
+      - name: Checkout circomlib
```

Comment:
how come this is not something we do while working locally but it's needed here?

Reply:
he step to manually checkout circomlib was introduced in April 2022, as part of a pull request (see [PR #403](https://github.com/privacy-scaling-explorations/maci/pull/403/files)). The reason for this step was due to a specific requirement at the time. maci-circuits imports circomlib and pins it to a specific commit (as indicated [here in the package.json](https://github.com/privacy-scaling-explorations/maci/blob/dev/circuits/package.json#L25)). Back then, the direct pulling of this dependency with a pinned commit like https://github.com/weijiekoh/circomlib#ac85e82c1914d47789e2032fb11ceb2cfdd38a2b failed.

This issue could have been related to the behavior of npm.js or changes in GitHub's security policy, as outlined in this [GitHub blog post](https://github.blog/2021-09-01-improving-git-protocol-security-github/) or specific to the GitHub Actions Runner environment.

However, I've re-triggered the CI run without this manual checkout step and it worked smoothly, indicating that this step may no longer be necessary. It seems that the underlying issue has been resolved, either through updates to npm.js or GitHub's handling of dependencies.